app/controllers/submissions_controller.rb

```diff
@@ -14,6 +14,7 @@ def page_title
   end
 
   def index
+    @submissions.each(&:rerun_tests)
```

Comment:
this is indeed not a great solution. What is the current problem with the custom path? Are you still getting errors? I'll try to make some time to replicate this myself.